lib/mocha-tests-lint.ts

```diff
@@ -0,0 +1,50 @@
+import * as Bluebird from 'bluebird';
+import * as fs from 'fs';
+
+// Mocha tests linter results.
+// When isError is set to true, the message contains details about found errors.
+export class MochaListResult {
+	constructor(
+		public readonly message: string,
+		public readonly isError: boolean,
+	) {}
+}
+
+const checkPattern = /(describe|it)\.only/;
+
+// Check if files in the test directory don't contain describe.only or it.only statements
+// to ensure that an accidental commit does not prevent all the tests from running.
+export async function lintMochaTests(
+	scripts: string[],
+): Promise<MochaListResult> {
+	let errorsFound = false;
+	let message = '';
+
+	const readFile = Bluebird.Promise.promisify(fs.readFile);
+
+	const allChecks = [];
+	for (const scriptPath of scripts) {
```

Comment:
You can do this as
```ts
const allChecks = scripts.map(async (script) => {
	content = await readFile(scriptPath, 'utf8')
	...
})

Reply:
Speaking about `readFile(scriptPath, 'utf8')`, it does not compile. Type of `Bluebird.promisify(fs.readFile)` allows only one argument.

Reply:
Yeah, that's a typing issue because if you did
```ts
const readFile = Bluebird.Promise.promisify(fs.readFile) as any;
```
then it would work fine at runtime.  I normally use `fs.promises.readFile` in node 10+ to get around this because it has the built in types that work nicely but we can't guarantee that for resin-lint :(